### DISCUSSION TITLE
array of enum

### DISCUSSION THREAD

**Discussion by apologeticasteroid** [prisma/prisma#18560] [Q&A] [answered]
How does one create an array of enum type on a model? Enums are apparently to be treated like scalars, but I don't know what is going wrong. Can someone tell me, is my code faulty?
example schema:
```prisma
model User {
  id       String         @id @default(cuid())
  email    String         @unique
  password Password?
  roletype UserRoleType[] <--- THIS PART I don't understand
}

enum UserRoleType {
  ADMIN
  USER
  UNASSIGNED
}
```
example seeder for the user:
```ts
await prisma.user.create({
    data: {
      email,
      password: {
        create: {
          hash: hashedPassword,
        },
      },
      roletype: [] <--- THIS PART I don't understand
    },
  });
```

**Comment by ludralph** [comment, marked as answer]
Hi @apologeticasteroid 👋 

To create a new user with one or more roles, you can pass an array of UserRoleType values when creating the user record. Here is an example of how to create an array of enum type in Prisma ORM
```js
await prisma.user.create({
  data: {
    email: 'user@example.com',
    password: {
      create: {
        hash: 'passwordHash',
      },
    },
    roletype: [UserRoleType.ADMIN, UserRoleType.USER],
  },
});
```
In the above example, a new user is created with an email and password, and two roles are assigned to the user: ADMIN and USER. The `roletype` field is initialized with an array containing the two enum values.

**Reply by ludralph** [reply]
Hi @apologeticasteroid 

Do you have any more questions around this issue? If no, it would be really helpful if you could quickly **mark this Discussion as _answered_** to indicate that your issue has been resolved. 

In case you do have any outstanding questions, feel free to let us know and we'd be happy to help!

**Reply by Benhidayat** [reply]
Hi, I'm not the original poster. Thanks for the solution since i'm facing the same issue. 

i have further question. if i wanted to set a default value for the role is this gonna be work? 
roletype UserRoleType[] @default([USER])
